lib/node_modules/@stdlib/array/base/reshape/test/test.js

```diff
@@ -0,0 +1,337 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var reshape = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof reshape, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns a reshaped nested array (2d, lexicographic)', function test( t ) {
```

Comment:
All your tests here and below assume the same number of dimensions. You need to include additional tests for reshaping arrays to nested arrays having different numbers of dimensions. E.g., 2d to 1d, 2d to 3d, etc. You don't need to be completely exhaustive, as the number of dimensions is unbounded, but you should at least demonstrate this for each array rank (2d, 3d, 4d, 5d, etc), which you've tested below.